### PR TITLE
UnitTests

### PR DESCRIPTION
**Pull request by bramberry**

### CODE REVIEW

File: src/test/java/com/example/simple/rest/AuthServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class AuthServiceTests {
+    @Autowired
+    AuthService authService;
+
+
+    @Test
+    public void createUser_ReturnNewUser() throws IOException {
+        User testUser = new User("aaaaa", DigestUtils.sha256Hex("adminadmin"), false);
```
Comment (DenSev) [inline src/test/java/com/example/simple/rest/AuthServiceTests.java]: General structure usually goes like this:

```java
         //given
         User expectedUser = new User("aaaaa", DigestUtils.sha256Hex("adminadmin"), false); 
         //when
         User actualUser = authService.createUser("aaaaa", "adminadmin")
         //then
         assertEquals(actualUser, expectedUser); 
```

File: src/test/java/com/example/simple/rest/AuthServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        assertEquals(testUser, authService.createUser("aaaaa", "adminadmin"));
+    }
+
+    @Test
+    public void checkAdmin_ReturnTrue() {
```
Comment (DenSev) [inline src/test/java/com/example/simple/rest/AuthServiceTests.java]: This style of test names is fine, but you should use established naming conventions when writing tests in future. What I mean is that when customer's codebase uses names like `testCheckAdminReturnTrue` - you shouldn't name your own test `checkAdmin_ReturnTrue`.

File: src/test/java/com/example/simple/rest/AuthServiceTests.java (hunk 1 above)
Comment (bramberry) [reply src/test/java/com/example/simple/rest/AuthServiceTests.java]: Is it necessary to write comments like "//{text}"?

File: src/test/java/com/example/simple/rest/CredentialsUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @Test
+    public void isValid_PasswordNull_ReturnFalse() {
+        assertFalse(CredentialsUtil.isValid("asd", null));
+    }
+
+    @Test
+    public void isValid_UsernameLengthLess_ReturnFalse() {
+        assertFalse(CredentialsUtil.isValid("asd", "asd"));
```
Comment (DenSev) [inline src/test/java/com/example/simple/rest/CredentialsUtilTest.java]: For tests checking the same functionality and with similar asserts data providers can be used. I'm not sure if jUnit has functionality similar to testNg @DataProvider, but keep this in mind for the future.

File: src/test/java/com/example/simple/rest/AuthServiceTests.java (hunk 1 above)
Comment (DenSev) [reply src/test/java/com/example/simple/rest/AuthServiceTests.java]: Not necessary, but encouraged. If you don't write comments like this then at least leave a blank line between three main blocks: preparing test data, calling tested method, asserting results.

**Pull request merged**